mcp/src/databricks_mcp/mcp.py

```diff
@@ -0,0 +1,130 @@
+            async with ClientSession(read_stream, write_stream) as session:
+                await session.initialize()
+                return (await session.list_tools()).tools
+
+    async def _call_tools_async(
+        self,
+        tool_name: str,
+        arguments: dict[str, Any] | None = None,
+    ) -> CallToolResult:
+        """Call the tool with the given name and input."""
+        async with streamablehttp_client(
+            url=self.mcp_url,
+            auth=DatabricksOAuthClientProvider(self.client),
+        ) as (read_stream, write_stream, _):
+            async with ClientSession(read_stream, write_stream) as session:
+                return await session.call_tool(tool_name, arguments)
+
+    def _extract_genie_id(self) -> str:
+        """Extract the Genie space ID from the URL."""
+        path = urlparse(self.mcp_url).path
+        if "/genie/" not in path:
+            raise ValueError(f"Missing /genie/ segment in: {self.mcp_url}")
+        genie_id = path.split("/genie/", 1)[1]
+        if not genie_id:
+            raise ValueError(f"Genie ID not found in: {self.mcp_url}")
+        return genie_id
+
+    def _normalize_tool_name(self, name: str) -> str:
+        """Convert double underscores to dots for compatibility."""
+        return name.replace("__", ".")
+
+    def list_tools(self) -> List[Tool]:
+        """List the tools from the MCP URL."""
+        return asyncio.run(self._get_tools_async())
+
+    def call_tool(self, tool_name: str, arguments: dict[str, Any] | None = None) -> CallToolResult:
+        """Call the tool with the given name and input."""
+        return asyncio.run(self._call_tools_async(tool_name, arguments))
+
+    def get_databricks_resources(self) -> List[DatabricksResource]:
+        """
+        This method is used to get the Databricks resources from the MCP URL.
+        It will return a list of Databricks resources.
+
+        Note that this method is only for databricks managed MCP URLs. It returns empty list for other MCP URLs
+        """
+        try:
+            if not self._is_valid_databricks_managed_mcp_url():
+                logger.warning(f"Ignoring invalid MCP URL: {self.mcp_url}")
+                return []
+
+            mcp_type = self._get_databricks_managed_mcp_url_type()
+
+            if mcp_type == GENIE_MCP:
+                return [DatabricksGenieSpace(self.extract_genie_id())]
+
+            tools = self.list_tools()
+            normalized = [self._normalize_tool_name(tool.name) for tool in tools]
+
+            if mcp_type == UC_FUNCTIONS_MCP:
+                return [DatabricksFunction(name) for name in normalized]
+            elif mcp_type == VECTOR_SEARCH_MCP:
+                return [DatabricksVectorSearchIndex(name) for name in normalized]
+
+            logger.warning(f"Unknown MCP type: {mcp_type}")
+            return []
+
+        except ValueError as ve:
+            logger.warning(f"Failed to parse MCP URL: {ve}")
```

Comment:
```suggestion
            logger.exception(f"Failed to parse MCP server URL {self.server_url}, returning empty list of resources")
```

This will cause a more detailed/useful stacktrace to be printed

Reply:
Though pretty sure we don't need this ValueError handling if we make the change suggested above (handle the case of "this URL doesn't match any known Databricks-managed MCP server URL once)